### ISSUE TITLE
add usage examples to most documentation

### ISSUE THREAD

**Issue by AboudyKreidieh** [flow-project/flow#51]
We need this if people are going to have a shot at understanding our codebase without constantly asking us every time they explore something new.

Example:
```
class RLController(BaseController):
    """Instantiates an RL Controller.

    Vehicle with this class specified will be stored in the list of the RL IDs
    in the Vehicles class.

    Usage:

        >>> from flow.core.vehicles import Vehicles
        >>> vehicles = Vehicles()
        >>> vehicles.add(acceleration_controller=(RLController, {}))

    In order to collect the list of all RL vehicles in the next, run:

        >>> rl_ids = vehicles.get_rl_ids()

    Attributes
    ----------
    veh_id: str
        Vehicle ID for SUMO identification
    fail_safe: str, optional
        type of flow-imposed failsafe the vehicle should posses, defaults
        to no failsafe (None)
    """
```

**Comment by kjang96** [comment]
@mtgibson2014  Start by figuring out what needs more documentation

**Comment by mtgibson2014** [comment]
It seems like we need to talk about what we expect our users to use in their own experiments. I'm guessing all environments, human controllers, traffic scenarios, and visualizers? 

Also, to this end, it's actually unclear about "how" users use Flow and what they can use it for. Aside from the examples **SHOWING** users how to use Flow, there's nothing that clearly shows what they are suppose to use for their own experiment (what's the specific API/ putting everything together). Yes, we do have the tutorials, but they are very detailed and there's no "intro" to the general idea about what users are expected to do with Flow. 

I talked to Kathy and we thought that maybe it'll be a good idea to put a general "Usage Example" in the ReadMe - with pictures of examples of code.

**Comment by AboudyKreidieh** [comment]
resolved by #679
